inc/cleantalk-settings.php

```diff
@@ -2679,3 +2684,21 @@ class="apbct-btn-as-link"
                 . __('change email', 'cleantalk-spam-protect') .
             '</button>)';
 }
+
+/**
+ * Implementation of service_update_local_settings functionality
+ */
+add_action('apbct_before_returning_settings', 'apbct__send_local_settings_to_api');
+
+function apbct__send_local_settings_to_api($settings)
+{
+    global $apbct;
+
+    // Settings to JSON
+    $settings = json_encode($settings);
+
+    // Hostname
+    $hostname = preg_replace('/^(https?:)?(\/\/)?(www\.)?/', '', get_site_url());
+
+    \Cleantalk\Common\API::methodSendLocalSettings($apbct->api_key, $hostname, $settings);
```

Comment:
Make an asynchronous remote call here instead of the direct API call. This way will allow preventing delays on the saving settings and will not require adding async functionality to the API class.

Reply:
Will be developed on the next stage.